Cython/Compiler/Code.py

```diff
@@ -512,13 +512,17 @@ def load(cls, util_code_name, from_file, **kwargs):
                     continue
                 # only pass lists when we have to: most argument expect one value or None
                 if name == 'requires':
-                    if orig_kwargs:
-                        values = [cls.load(dep, from_file, **orig_kwargs)
-                                  for dep in sorted(values)]
-                    else:
-                        # dependencies are rarely unique, so use load_cached() when we can
-                        values = [cls.load_cached(dep, from_file)
-                                  for dep in sorted(values)]
+                    def load_dep(dep):
+                        if cls.is_cython_utility and re.search("[.]c(pp)?::", dep):
+                            # Allow loading of C utility code from Cython utility code.
+                            # Note that we don't current propagate things like tempita context.
+                            return UtilityCode.load_cached(dep, from_file)
+                        if orig_kwargs:
+                            return cls.load(dep, from_file, **orig_kwargs)
+                        else:
+                            return cls.load_cached(dep, from_file)
+
+                    values = [load_dep(dep) for dep in sorted(values)]
```

Comment:
Yeah, I had also looked into implementing something like this but didn't get very far. It seems better to leave the distinction to the `CythonUtilityCode` class. It can override `.load()` and delegate to the base class for unsupported file formats. That's better than adding a special case here that knows about three different classes of the type hierarchy.